src/crud.js

```diff
@@ -0,0 +1,191 @@
+  showToDos();
+};
+
+const saveTodo = (id, isBtn = false) => {
+  const taskList = document.querySelector(`#tasks-${id}`);
+  const saveEdit = document.querySelector(`.save-${id}`);
+  const deleteEdit = document.querySelector(`.delete-${id}`);
+  const inputId = document.querySelector(`#activity-${id}`);
+  const editBtn = document.getElementById(`${id}`);
+  editBtn.style.display = 'block';
+
+  const existingTodos = JSON.parse(localStorage.getItem('todos'));
+  existingTodos[id].description = inputId.value;
+  localStorage.setItem('todos', JSON.stringify(existingTodos));
+  if (isBtn) {
+    saveEdit.remove();
+    deleteEdit.remove();
+  }
+  taskList.classList.remove('active');
+  inputId.setAttribute('readonly', true);
+};
+
+const saveTodos = (e) => {
+  const saveBtn = e.target;
+  const btnClass = saveBtn.className;
+  const btnId = btnClass.split('-');
+  const id = parseInt(btnId[1], 10);
+  saveTodo(id, true);
+};
+
+const editTodos = (e) => {
+  const editInput = e.target;
+  const editBtn = document.getElementById(`${editInput.id}`);
+  const inputId = document.querySelector(`#activity-${editInput.id}`);
+  inputId.removeAttribute('readonly');
+  inputId.focus();
+
+  editBtn.style.display = 'none';
+  inputId.style.outlineColor = 'gray';
+
+  const taskList = document.querySelector(`#tasks-${editInput.id}`);
+  taskList.classList.add('active');
+
+  const task = document.querySelector(`#task-${editInput.id}`);
+
+  const save = document.createElement('img');
+  save.setAttribute('src', saveIcon);
+  save.classList.add(`save-${editInput.id}`);
+  task.appendChild(save);
+
+  const removeBtn = document.createElement('img');
+  removeBtn.classList.add(`delete-${editInput.id}`);
+  removeBtn.setAttribute('src', deleteIcon);
+  task.appendChild(removeBtn);
+
+  document.querySelectorAll(`.delete-${editInput.id}`).forEach((e) => {
+    e.addEventListener('click', deleteTodos);
+  });
+
+  document.querySelectorAll(`.save-${editInput.id}`).forEach((e) => {
+    e.addEventListener('click', saveTodos);
+  });
+};
+
+const showToDos = () => {
+  const existingTodos = JSON.parse(localStorage.getItem('todos'));
+  if (existingTodos !== null && existingTodos.length > 0) {
+    const collection = document.querySelector('.todo-collection');
+    collection.innerHTML = '';
+
+    existingTodos.forEach((todo, index) => {
+      const unordered = document.createElement('ul');
+      const ordered = document.createElement('li');
+      const liner = document.createElement('hr');
+
+      unordered.classList.add('tasks');
+      unordered.id = `tasks-${index}`;
+
+      ordered.classList.add('tasks');
+      ordered.id = `task-${index}`;
+
+      const check = document.createElement('input');
+      check.type = 'checkbox';
+      check.classList.add('check');
+      check.id = `check-${index}`;
+
+      const input = document.createElement('input');
+      input.type = 'text';
+      input.classList.add('activity');
```

Comment:
- Kindly consider storing all this in a separate function to make your code more neat and organized and the same will apply up to line 132. You could perhaps consider using string literals and store it in a separate file

Reply:
Thanks for the review. I didn't re-use these parts of my codes that is why I didn't store them in seperate variables. They were used only once.

Reply:
Ohhh !! Then it should be okay